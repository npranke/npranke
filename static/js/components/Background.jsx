import React from 'react'

import Landscape from '@img/background-landscape.jpg'
import Portrait from '@img/background-portrait.jpg'

function Background() {
    return (
        <picture>
            <source srcSet={ Portrait } media="(orientation: portrait)" />
            <img
                src={ Landscape }
                className="background"
                alt="Conifer in Colorado landscape"
            />
        </picture>
    )
}

export default Background
