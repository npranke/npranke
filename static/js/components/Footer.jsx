function Footer() {
    const currentYear = new Date().getFullYear()

    return (
        <footer className="footer" role="contentinfo">
            <div className="copyright">
                &copy; 2017-{currentYear} Nicole Pranke
            </div>
        </footer>
    )
}

export default Footer
