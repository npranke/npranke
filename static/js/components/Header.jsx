import { NavLink } from 'react-router-dom-v5-compat'
import PropTypes from 'prop-types'
import React from 'react'

import GitHub from '@img/icon-github.png'
import Home from '@img/icon-home.png'
import LinkedIn from '@img/icon-linkedin.png'
import Workbook from '@img/icon-workbook.png'

import { getSendEventHandler } from '@utils'

function Header(props) {
    const homeLink = props.isPageNotFound
        ? (
            <a
                onClick={ getSendEventHandler(
                    'header',
                    'navigate',
                    'home',
                ) }
                href="/home"
                target="_self"
                rel="noreferrer"
                title="Home"
                aria-label="Home"
            >
                <img src={ Home } className="icon icon-home" alt="Home icon" />
            </a>
        ) : (
            <NavLink
                end
                onClick={ getSendEventHandler(
                    'header',
                    'navigate',
                    'home',
                ) }
                to="/home"
                target="_self"
                rel="noreferrer"
                className="navlink navlink-home"
                title="Home"
                aria-label="Home"
            >
                <img src={ Home } className="icon icon-home" alt="Home icon" />
            </NavLink>
        )

    const workbookLink = props.isPageNotFound
        ? (
            <a
                onClick={ getSendEventHandler(
                    'header',
                    'navigate',
                    'workbook',
                ) }
                href="/workbook"
                target="_self"
                rel="noreferrer"
                title="Workbook"
                aria-label="Workbook"
            >
                <img
                    src={ Workbook }
                    className="icon icon-workbook"
                    alt="Workbook icon"
                />
            </a>
        ) : (
            <NavLink
                onClick={ getSendEventHandler(
                    'header',
                    'navigate',
                    'workbook',
                ) }
                to="/workbook"
                target="_self"
                rel="noreferrer"
                className="navlink navlink-workbook"
                title="Workbook"
                aria-label="Workbook"
            >
                <img
                    src={ Workbook }
                    className="icon icon-workbook"
                    alt="Workbook icon"
                />
            </NavLink>
        )

    return (
        <nav className="header">
            <div className="left-header">
                <div className="icon-container icon-container-home">
                    { homeLink }
                </div>
            </div>
            <div className="right-header">
                <div className="icon-container icon-container-workbook">
                    { workbookLink }
                </div>
                <div className="icon-container icon-container-github">
                    <a
                        onClick={ getSendEventHandler(
                            'header',
                            'navigate',
                            'github',
                        ) }
                        href="https://github.com/npranke"
                        target="_blank"
                        rel="noreferrer noopener external"
                        title="GitHub"
                        aria-label="GitHub"
                    >
                        <img
                            src={ GitHub }
                            className="icon icon-github"
                            alt="GitHub icon"
                        />
                    </a>
                </div>
                <div className="icon-container icon-container-linkedin">
                    <a
                        onClick={ getSendEventHandler(
                            'header',
                            'navigate',
                            'linkedin',
                        ) }
                        href="https://www.linkedin.com/in/npranke"
                        target="_blank"
                        rel="noreferrer noopener external"
                        title="LinkedIn"
                        aria-label="LinkedIn"
                    >
                        <img
                            src={ LinkedIn }
                            className="icon icon-linkedin"
                            alt="LinkedIn icon"
                        />
                    </a>
                </div>
            </div>
        </nav>
    )
}

Header.defaultProps = {
    isPageNotFound: false,
}

Header.propTypes = {
    isPageNotFound: PropTypes.bool,
}

export default Header
