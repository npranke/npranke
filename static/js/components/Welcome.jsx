import { Link } from 'react-router-dom'
import React from 'react'

import { getLocationPageTitle, getSendEventHandler } from '../utils'
import Workbook from '../../img/icon-workbook.png'

function Welcome() {
    document.title = getLocationPageTitle('home')

    return (
        <main className="welcome">
            <div className="intro">
                Welcome! <br />
                My name is Nicole; I&#8217;m a software engineer.
            </div>
            <div className="button-workbook">
                <Link
                    onClick={ getSendEventHandler(
                        'welcome',
                        'navigate',
                        'workbook',
                    ) }
                    to="/workbook"
                    target="_self"
                    rel="noreferrer"
                >
                    <img
                        src={ Workbook }
                        className="icon icon-workbook"
                        alt="Workbook icon"
                    />
                    <span className="text-workbook">Workbook</span>
                </Link>
            </div>
        </main>
    )
}

export default Welcome
