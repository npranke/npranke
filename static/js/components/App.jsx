import React from 'react'
import { Route, Switch } from 'react-router-dom'

import Background from './Background'
import Concentration from './Concentration'
import Footer from './Footer'
import Header from './Header'
import PageNotFound from './PageNotFound'
import Welcome from './Welcome'
import Workbook from './Workbook'

function App() {
    return (
        <div className="app">
            <Background />
            <Switch>
                <Route
                    exact
                    path={ [
                        '/',
                        '/home',
                        '/workbook',
                        '/workbook/concentration',
                    ] }
                    component={ Header }
                />
                <Route
                    render={
                        (props) => {
                            return <Header { ...props } isPageNotFound />
                        }
                    }
                />
            </Switch>
            <Switch>
                <Route exact path={ ['/', '/home'] } component={ Welcome } />
                <Route exact path="/workbook" component={ Workbook } />
                <Route
                    exact
                    path="/workbook/concentration"
                    component={ Concentration }
                />
                <Route component={ PageNotFound } />
            </Switch>
            <Footer />
        </div>
    )
}

export default App
