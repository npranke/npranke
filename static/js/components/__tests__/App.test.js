import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow } from 'enzyme'
import React from 'react'

import App from '../App'
import Background from '../Background'
import Footer from '../Footer'
import Header from '../Header'
import Welcome from '../Welcome'

Enzyme.configure({ adapter: new Adapter() })

describe('App', () => {
    test('contains background', () => {
        const app = shallow(<App />)

        expect(
            app.contains(<Background />),
        ).toBe(true)
    })

    test('contains header', () => {
        const app = shallow(<App />)

        expect(
            app.contains(<Header />),
        ).toBe(true)
    })

    test('contains welcome', () => {
        const app = shallow(<App />)

        expect(
            app.contains(<Welcome />),
        ).toBe(true)
    })

    test('contains footer', () => {
        const app = shallow(<App />)

        expect(
            app.contains(<Footer />),
        ).toBe(true)
    })
})

describe('App snapshot', () => {
    test('matches snapshot', () => {
        const app = shallow(<App />)

        expect(app).toMatchSnapshot()
    })
})
