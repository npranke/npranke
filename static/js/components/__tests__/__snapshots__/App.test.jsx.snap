// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot matches snapshot 1`] = `
<div
  className="text"
>
  Hello world of silly cats!
</div>
`;
